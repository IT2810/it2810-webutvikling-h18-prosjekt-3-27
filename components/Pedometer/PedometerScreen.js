import React from "react";
import { Pedometer } from "expo";
import { StyleSheet, Text, View } from "react-native";
import ProgressCircle from 'react-native-progress/Circle';

export default class PedometerScreen extends React.Component {
  state = {
    isPedometerAvailable: "checking",
    pastStepCount: 0,
    currentStepCount: 0
  };

  componentDidMount() {
    this._subscribe();
  }

  componentWillUnmount() {
    this._unsubscribe();
  }

  _subscribe = () => {
    this._subscription = Pedometer.watchStepCount(result => {
      this.setState({
        currentStepCount: result.steps
      });
    });

    Pedometer.isAvailableAsync().then(
      result => {
        this.setState({
          isPedometerAvailable: String(result)
        });
      },
      error => {
        this.setState({
          isPedometerAvailable: "Could not get isPedometerAvailable: " + error
        });
      }
    );

    const end = new Date();
    const start = new Date();
    start.setDate(end.getDate() - 1);
    Pedometer.getStepCountAsync(start, end).then(
      result => {
        this.setState({ pastStepCount: result.steps });
      },
      error => {
        this.setState({
          pastStepCount: "Could not get stepCount: " + error
        });
      }
    );
  };

  _unsubscribe = () => {
    this._subscription && this._subscription.remove();
    this._subscription = null;
  };

  getProgress = () => {
    const goal = 10000;
    let count = this.state.pastStepCount;
    let progress = count / goal;
    return progress;
  };

  render() {
    return (
      <View style={styles.container}>
        <Text>
          Pedometer.isAvailableAsync(): {this.state.isPedometerAvailable}
        </Text>
        <Text>
         You have walked {this.state.pastStepCount} steps today!
        </Text>
        <Text>Live count: {this.state.currentStepCount}</Text>

        <ProgressCircle progress={isNaN(this.getProgress()) ? 0 : this.getProgress()} size={100} showsText={true}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 15,
    alignItems: "center",
    justifyContent: "center"
  }
});