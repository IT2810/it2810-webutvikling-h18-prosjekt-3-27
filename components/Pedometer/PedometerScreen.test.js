import React from 'react';
import renderer from 'react-test-renderer';

import PedometerScreen from "./PedometerScreen";


test('renders correctly', () => {
  const tree = renderer.create(<PedometerScreen/>).toJSON();
  expect(tree).toMatchSnapshot();
});
