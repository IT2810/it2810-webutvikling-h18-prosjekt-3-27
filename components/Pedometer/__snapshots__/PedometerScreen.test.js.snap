// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 25,
          }
        }
      >
        You have walked
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 40,
          }
        }
      >
        0
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        of 10000 steps today!
      </Text>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <ARTSurfaceView
          style={
            Array [
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              Object {
                "height": 250,
                "width": 250,
              },
            ]
          }
        >
          <ARTShape
            d={
              Array [
                0,
                125,
                6,
                2,
                125,
                6,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.8980392156862745,
                0.00392156862745098,
                0,
                1,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={10}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
          <ARTShape
            d={
              Array [
                0,
                125,
                0.5,
                4,
                125,
                125,
                124.5,
                -1.5707963267948966,
                1.5707963267948966,
                1,
                4,
                125,
                125,
                124.5,
                1.5707963267948966,
                -1.5707963267948966,
                1,
                1,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.8980392156862745,
                0.00392156862745098,
                0,
                1,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={1}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
        </ARTSurfaceView>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 114,
              "height": 228,
              "justifyContent": "center",
              "left": 11,
              "position": "absolute",
              "top": 11,
              "width": 228,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#E50100",
                  "fontSize": 50.666666666666664,
                  "fontWeight": "300",
                },
                undefined,
              ]
            }
          >
            0%
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
